compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -957,6 +957,17 @@ impl<'a> Parser<'a> {
             }
         }
 
+        if self.token.kind == TokenKind::EqEq
+            && self.prev_token.is_ident()
+            && expected.iter().any(|tok| matches!(tok, TokenType::Token(TokenKind::Eq)))
+        {
+            // Likely typo: `=` → `==` in let expr or enum item
+            let mut err = self.struct_span_err(self.token.span, "expect `=`, found `==`");
+            err.span_suggestion_short(self.token.span, "write `=` instead of `==`", "=", appl)
```

Comment:
```suggestion
            let mut err = self.struct_span_err(self.token.span, "unexpected `==`");
            err.span_suggestion_short(self.token.span, "try using `=` instead", "=", appl)
```

Reply:
`SessionDiagnostic` will print the suggestion in the same line, so it will print like this:

```console
error: unexpected `==`
 --> src/test/ui/parser/issue-101477-enum.rs:6:7
  |
6 |     B == 2 //~ ERROR unexpected `==`
  |       ^^ help: try using `=` instead `==`: `=`
```

I think it's better in this format?

```console
error: unexpected `==`
 --> src/test/ui/parser/issue-101477-let.rs:4:11
  |
4 |     let x == 2; //~ ERROR unexpected `==`
  |           ^^ help: replace `==` with a equal symbol: `=`
```

Reply:
Resolved with @compiler-errors with advices.